### DISCUSSION TITLE
How to use standalone dashboard with resources

### DISCUSSION THREAD

**Discussion by mguirao** [microsoft/aspire#4440] [Q&A] [answered]
Hey,
I'm looking for directions and documentation on configuring the `Dashboard:ResourceServiceClient:Url` endpoint for my standalone dashboard running in Kubernetes.

From the documentation, I'm supposed to fill the configuration with `The gRPC endpoint to which the dashboard connects for its data.`.
How can I make this work ?

Thanks in advance

**Comment by davidfowl** [comment]
Somebody needs to build a resource server. There's only one implementation right now and it's the one that works with aspire when you run with the orchestration pieces. The standalone dashboard doesn't have a resource server to talk to. To make it work you have to build a grpc server that implements this contract https://github.com/dotnet/aspire/blob/main/src/Aspire.Hosting/Dashboard/proto/resource_service.proto.

**Comment by pksorensen** [comment]
@davidfowl - where would the list of possible commands that the dashboard might send according to the protobuf spec be ? :D Possible that there is some documentation of that too.

**Reply by pksorensen** [reply]
Woops, never mind that was part of the model also of the resources

**Reply by pksorensen** [reply]
seems not that hard, 
![image](https://github.com/user-attachments/assets/6979f576-bc3c-4fd7-a0c2-4e2559e3a7e7)
![image](https://github.com/user-attachments/assets/607bb35d-e39e-413b-b12f-f9b8e02a62a5)
now what to do with it 🚀

**Reply by pksorensen** [reply]
quick dashboard over containers running on my docker desktop: 
![image](https://github.com/user-attachments/assets/bd202c83-eb43-46fa-9caa-240974b30e16)

**Reply by davidfowl** [reply]
Beautiful!

**Reply by brettwinters** [reply]
Hi @pksorensen - Well done! please share your code - or even better, a resource service middleware which I can integrate into one of my services

**Reply by davidfowl** [reply]
@pksorensen Is your code open source?

**Reply by swimmesberger** [reply]
@pksorensen would also be interested in a docker based resource server impl.

**Reply by pksorensen** [reply]
Sorry - missed the replies.  I can make the code open source but bear in mind that it was just a two hour proof of concept and not a full implementation - kinda parked  it until resource commands hits in 9.0 :) 

I will dig up the code in the following week

**Reply by kiapanahi** [reply]
Hey there @pksorensen 

I also did something similar and was about to open-source it this weekend. My solution is also quick and dirty ATM.

I'd be happy to contribute to your project if you're up for it :)

**Reply by Wraiyth** [reply]
I've just come across this post looking for ways to be able to customize the resources that are returned for the built-in Dashboard. Having a way to be able to customize that would be awesome (intercepting the gRPC calls to get the resources somehow?) so you can customize the resource view a bit further

**Reply by davidfowl** [reply]
Sounds like an excellent community project idea!

**Reply by HClausing** [reply]
> Sounds like an excellent community project idea!

I'm very interested to contribuite on it.

I've created a test project [here](https://github.com/Eficaz-Sistemas/EficazFramework.Hosting)

I'd happy to receive contributions here by those how are interested.

**Edit**
@kiapanahi 's project is at more advanced stage.

**Reply by kiapanahi** [reply]
Hey guys,

Sorry for not updating this place as fast as I should have.

I have pushed my fundamental implementation of the standalone resource server [here](https://github.com/kiapanahi/Aspire.ResourceServer.Standalone).

@HClausing I think this one is a tad further in the implementation phase. Please review it and tell me whether you'd like to move it to your repository or collaborate on it.

P.S.
I have attached a screenshot for you to see what is currently being populated in the dashboard

![image](https://github.com/user-attachments/assets/551d1f0d-8b3f-4b58-87b9-15f403db42b8)

**Reply by davidfowl** [reply]
I love this! @kiapanahi can you make this an announcement on the aspire discussions? Lets see if we can get people interested in this! This is great for the ecosystem!

**Reply by kiapanahi** [reply]
@davidfowl sure thing. I'll tend to it ASAP.

**Reply by HClausing** [reply]
> the aspire discussions? Lets see if we can get people interested in this! This is great for the ecosystem!

Amazing! If you agree to, I would like to contribute on yours. Mine is just an initial experiment.

**Reply by kiapanahi** [reply]
> > the aspire discussions? Lets see if we can get people interested in this! This is great for the ecosystem!
> 
> Amazing! If you agree to, I would like to contribute on yours. Mine is just an initial experiment.

That'll be my honor and I'd appreciate your insights.

**Reply by kiapanahi** [reply, marked as answer]
@davidfowl #6772 is created as per your suggestion. Since this was my first time opening a discussion I rely on your (and others) helpful comments or ideas if you think it needs improvements.
